### ISSUE TITLE
bob: Update to clarify ambiguity regarding shouted questions

### ISSUE THREAD

**Issue by kytrinyx** [exercism/cfml#74]
**TL;DR**: the problem specification for the Bob exercise has been updated. Consider updating the test suite for Bob to match. If you decide not to update the exercise, consider overriding description.md.

-----

**Details**

The problem description for the Bob exercise lists four conditions:

- asking a question
- shouting
- remaining silent
- anything else

There's an ambiguity, however, for shouted questions: should they receive the "asking" response or the "shouting" response?

In https://github.com/exercism/problem-specifications/pull/1025 this ambiguity was resolved by adding an additional rule for shouted questions.

If this track uses exercise generators to update test suites based on the canonical-data.json file from problem-specifications, then now would be a good time to regenerate 'bob'. If not, then it will require a manual update to the test case with input "WHAT THE HELL WERE YOU THINKING?".

See the [most recent canonical-data.json file](https://github.com/exercism/problem-specifications/blob/6055eb4b103e06c18184287b2be48f3c870aadcf/exercises/bob/canonical-data.json#L53-L58) for the exact changes.

Remember to regenerate the exercise README after updating the test suite:

    configlet generate . --only=bob --spec-path=<path to your local copy of the problem-specifications repository>

You can download the most recent configlet at https://github.com/exercism/configlet/releases/latest if you don't have it.

If, as track maintainers, you decide that you don't want to change the exercise, then please consider copying [problem-specifications/exercises/bob/description.md](https://github.com/exercism/problem-specifications/blob/8793a059fe18618fffeb97630baf76eebd5bf9a8/exercises/bob/description.md) into this track, putting it in `exercises/bob/.meta/description.md` and updating the description to match the current implementation. This will let us run the configlet README generation without having to worry about the bob README drifting from the implementation.

**Comment by kytrinyx** [comment]
I've submitted a PR to temporarily lock the description of 'bob' so that we can regenerate the exercise READMEs without worrying about getting an out-of-sync description.

This means that the exercises/bob/.meta/description.md file will need to be deleted if the test suite is updated to include the new rule.

**Comment by bdw429s** [comment]
@kytrinyx 

I went to regenerate the tests but I see my test generator doesn't work with the new format of canonical data.
https://github.com/exercism/problem-specifications/commit/6dc2014c946be9461649635fdbf89d84cd2aa8ca#diff-dc41ba274d2b92436918160694c5dbdd
Is this change applied for all exercises, or do I need to support both formats?  Also, what is the meaning of the `heybob` key in the input object now?  Is that meant to be the name of the parameter to the SUT?  I had always just used `input` as the parameter name.  I didn't see a link in the commit message from above that explained and I must have missed the discussion on it.

**Comment by bdw429s** [comment]
Oops, here it is.  I missed it:
https://github.com/exercism/problem-specifications/issues/996

Actually makes perfect sense.  I remember that being a little wonky when I wrote the test generator.  I'll need to go through each of the solutions and class stubs to update them now that we have meaningful parameter(s) name(s) on the methods.

**Comment by bdw429s** [comment]
Ouch, I guess the answer is no, they're not all updated yet.

https://github.com/exercism/problem-specifications/blob/master/exercises/grains/canonical-data.json

So I guess I need to support the newer and older versions of canonical data at the same time?

**Comment by Insti** [comment]
Or wait until they are all converted, or help out with that process..

**Comment by kytrinyx** [comment]
I've not been involved in the process of getting that all sorted out, but it sounds like you got to the bottom of it

> Or wait until they are all converted, or help out with that process..

🌸 Or that :)

**Comment by bdw429s** [comment]
Yep, I just didn't realize it was in progress at first.  It's not a big deal, I think I have my test generator working now with both formats which is fine.  I really love just being able to regen all my tests!  As soon as i work through all the solutions to update the changed parameter names, I'll send over a pull which will fix the Bob test cases as well as update all the tests with the new canonical data.

**Comment by bdw429s** [comment]
This pull updates the Bob test suite and I regenerated the readme with this:
```
box task run taskFile=generatereadme.cfc :slug=bob :force=true
```
